src/sources/soundsourceffmpeg.cpp

```diff
@@ -8,6 +8,21 @@
 
 namespace Mixxx {
 
+//static
+void SoundSourceProviderFFmpeg::init() {
+    av_register_all();
+    avcodec_register_all();
+}
+
+SoundSourceProviderFFmpeg::SoundSourceProviderFFmpeg() {
+    // NOTE(uklotzde): init() needs to be called only once.
+    // But since only a single instance of this class will
+    // be created it is sufficient to place the invocation
+    // of init() in the the constructor. This ensures that
+    // it will be called at least once.
+    init();
```

Comment:
Because there are 2 different aspects
* The code itself
* The invocation of this code

I wanted to separate the one-time-initialization code from the constructor. 
The keyword 'static' at the function declaration explicitly states, that 
this code has nothing to do with any instance. The constructor just invokes 
this code and explains why we decided to call this static initialization 
code from here.

On 01/07/2016 11:57 AM, kain88-de wrote:
>
> In src/sources/soundsourceffmpeg.cpp 
> <https://github.com/mixxxdj/mixxx/pull/842#discussion_r49062237>:
>
> > @@ -8,6 +8,21 @@
> >
> >  namespace Mixxx {
> >
> > +//static
> > +void SoundSourceProviderFFmpeg::init() {
> > +    av_register_all();
> > +    avcodec_register_all();
> > +}
> > +
> > +SoundSourceProviderFFmpeg::SoundSourceProviderFFmpeg() {
> > +    // NOTE(uklotzde): init() needs to be called only once.
> > +    // But since only a single instance of this class will
> > +    // be created it is sufficient to place the invocation
> > +    // of init() in the the constructor. This ensures that
> > +    // it will be called at least once.
> > +    init();
>
> why don't you place the function calls in the constructor?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/842/files#r49062237>.
>